### PR TITLE
Add devcontainer

### PR DESCRIPTION
**Pull request by Froxcey**
Long time no see, AvdanOS. 

This is PR focuses on DX, and no actual code is changed. Since this project can only run on Linux, I added the devcontainer configuration to easily launch this (VSCode) project in a container environment. If this PR gets merged, this project can be opened and tested on any operating system with Docker (and VSC) installed.

vibe

### CODE REVIEW

**Comment by MrSerge01** [comment]
FROXCEY IS BACK!!!!!!!!!!!!!!!!!!!

**Review by MrSerge01 — APPROVED** [review]
FROXCEY IS BACK!!!!!!11111111111

one small nitpick: you `.gitignore`'d `.devcontainer` but its still in the commit, weird
other than that, LGTM,
-The Goos.

**Comment by Froxcey** [comment]
@MrSerge01 

Yes, I got confused. Should be fixed now.

vibe

File: .devcontainer/pl10k.zsh [hunk 1]
```diff
@@ -0,0 +1,1647 @@
+# Generated by Powerlevel10k configuration wizard on 2023-01-26 at 06:37 UTC.
```
Comment (Froxcey) [inline .devcontainer/pl10k.zsh:1]: Maybe I should remove all the pl10k comments?

Comment (MrSerge01) [reply .devcontainer/pl10k.zsh:1]: yeah

**Comment by Sammy99jsp** [comment]
Huh, do Wayland Compositors work under Windows ?

**Review by Sammy99jsp — COMMENTED** [review]
LGTM ! 

Someone should try running `weston`, `anvil`, or this on WSL2, lol.

**Comment by Froxcey** [comment]
No, but it does compile, not run but compile, on Ubuntu running in Docker container on Windows or MacOS.

**Review by Sammy99jsp — APPROVED** [review]
Bruh, wrong checkbox.

**Comment by SunightMC** [comment]
> LGTM !
> 
> Someone should try running `weston`, `anvil`, or this on WSL2, lol.

Should I try this? :kekw:

**Pull request merged**